notes_app/client/app/notes/notes_service.js

```diff
@@ -1,19 +1,38 @@
 (function() {
   'use strict';
 
-  notesService.$inject = ['$http', '$log', '$q'];
+  notesService.$inject = ['$http', '$log'];
 
-  function notesService($http, $log, $q) {
+  function notesService($http, $log) {
     var api_path = 'http://localhost:8000/api/v1/note/';
     return {
       'list': function() {
-        return $http.get(api_path);
+        return $http.get(api_path).then(function successHandler(response) {
+          return response.data;
+        }).catch(function errorHandler(response) {
+          $log.error('Problem with getting the user list from server')
+          return {};
+        });
       },
       'read': function(id) {
-        return $http.get(api_path + id + '/');
+        return $http.get(api_path + id + '/').then(
+          function successHandler(response) {
+            return response.data;
+          }
+        ).catch(function errorHandler(response) {
+          $log.error('Problem with getting the user from server');
+          return {};
```

Comment:
I think here empty object is confusing. Code will wait an object, and if there is no object - it's better to return null, what do you think?